Session management

BJordi opened the pull request:
## Summary

- Added local storage service.
- Added methods to save and remove access token in the local storage.
- Added Auth view.
- Added logout method.
- Installed pug and pug-plain-loader libraries.

Commit:
Added local storage service

Commit:
Added methods to save access token in the local storage on successful login.

Commit:
Installed pug and pug-plain-loader libraries.

Commit:
Auth view
- Added auth view
- Added auth to router

Commit:
Logout
- Added functions to clear the access token in the local storage and set the basic header in the api configuration.

FrankIglesias commented on src/router.js:
```diff
@@ -18,6 +18,11 @@ export default new Router({
       component: Login
     },
+    {
+      path: '/auth',
```
`/`

BJordi replied:
I don think it should be `/`. The Trello says two things:
1) When the user goes to `/`, must be redirected to `/login`
2) The Trello card says this:
![image](https://user-images.githubusercontent.com/22583914/51263683-85f7b080-1993-11e9-9faf-42ea29235210.png)
That `Auth` component is actually the navbar. I already changed its name in the next PR and created a dashboard component that uses it.

FrankIglesias replied:
Yes, there was a mistake with the trello bullet. It's already been corrected

FrankIglesias commented on src/router.js:
```diff
@@ -18,6 +18,11 @@ export default new Router({
     },
+    {
+      path: '/auth',
+      name: 'auth',
```
dashboard

BJordi replied:
Read the comment above.

FrankIglesias commented on src/router.js:
```diff
@@ -18,6 +18,11 @@ export default new Router({
+    {
+      path: '/auth',
+      name: 'auth',
+      component: () => import('./views/Auth.vue')
```
rename the component to dashboard

BJordi replied:
Read two comments above.

FrankIglesias commented on src/views/Auth.vue:
```diff
@@ -0,0 +1,54 @@
+      img.logo(src='../assets/wolox-logo.png')
+      button.logout-button(@click='logout')
+        | Logout
+
```
delete this line

FrankIglesias commented on src/views/Auth.vue:
```diff
@@ -0,0 +1,54 @@
+<template lang="pug">
+  .nav-bar
+    .items-container
+      img.logo(src='../assets/wolox-logo.png')
```
add an alt

FrankIglesias commented on src/views/Auth.vue:
```diff
@@ -0,0 +1,54 @@
+import { deleteCurrentSession } from '@/services/auth'
+
+export default {
+  name: 'auth',
```
dashboard

FrankIglesias commented on src/views/Auth.vue:
```diff
@@ -0,0 +1,54 @@
+    display: flex;
+    height: 100%;
+    justify-content: space-between;
+    margin: 0 auto;
```
why this auto?

BJordi replied:
As I set a max-width, I want this component to be centered without adding a `display:flex; justify-content: center;` to its parent.

FrankIglesias replied:
It's recommendable to set alignments in the parent node

FrankIglesias commented on src/views/Auth.vue:
```diff
@@ -0,0 +1,54 @@
+  background-color: $white;
+  border-top: 5px solid $scooter;
+  box-shadow: 0 3px 5px $shadow;
+  width: 100vw;
```
100%?

BJordi replied:
Its parent is the div#app, which is being modified just by the scss `* { ... }`. Do you prefer I add a class to that div with a `width: 100%` and change this one to the same thing? Or should I leave it as it is now?

FrankIglesias commented on src/views/Auth.vue:
```diff
@@ -0,0 +1,54 @@
+    justify-content: space-between;
+    margin: 0 auto;
+    max-height: 50px;
+    max-width: 1080px;
```
why there's such a big max-width?

Commit:
DASHBOARD

- The Auth component was renamed to Dashboard, as there was an mistake in the Trello card.
Also, the navbar was extract from it into a new homonymous component.

Commit:
NAVBAR

- Fix styles in navbar. Removed unnecessary max-width and replaced it by paddings in the parent element.
- Fix redirection after login, now it redirects to dashboard instead of to auth.

Commit:
ROUTER

Solved conflicts with master branch.

mcallegari10 commented on src/views/Dashboard.vue:
```diff
@@ -0,0 +1,29 @@
+  display: flex;
+  flex-direction: column;
+  background-color: $light-grey;
+  height: 100vh;
```
No need for this

BJordi replied:
Actually, I need it or the container will take the height of its children, in this case, the navbar. In the tree, above this container there's only one more div, the #app one. Should I apply this height to that div and then use a `height: 100%` in this one?

mcallegari10 replied:
It's OK for the container in this case to have the same height as the children, so leave it in `auto` :smile:

mcallegari10 commented on src/views/Dashboard.vue:
```diff
@@ -0,0 +1,29 @@
+</script>
+
+<style lang="scss" scoped>
+@import '@/scss/commons/colors.scss';
```
Does the `@` work? I think it's just a shorthand for JS, not SCSS :thinking: . BTW, no need for the extension

BJordi replied:
If I delete the `@`, it doesn't work. Same goes for the extension. Both fail in the compilation :thinking:

mcallegari10 replied:
But you are using the relative path in the navbar component and it works just fine

mcallegari10 commented on src/views/Dashboard.vue:
```diff
@@ -0,0 +1,29 @@
+</template>
+
+<script>
+import NavBar from './NavBar'
```
Navbar should be in the `components` folder, not in views

BJordi replied:
Yeah, was thinking the same but wasn't sure about the structure. Will create that folder and move this component to it.

mcallegari10 commented on src/views/NavBar.vue:
```diff
@@ -0,0 +1,49 @@
+    img.navbar-logo(src='../assets/wolox-logo.png')
+    button.logout-button(@click='logout')
+      | Logout
+
```
No need for this blank line

mcallegari10 commented on src/views/NavBar.vue:
```diff
@@ -0,0 +1,49 @@
+import { deleteCurrentSession } from '@/services/auth'
+
+export default {
+  name: 'navBar',
```
No need for the name in this component

BJordi replied:
Why is it? :thinking:

mcallegari10 replied:
The only components in need of a name are the `views` since otherwise, you'll only see them as anonymous. Since you are importing the components with a name, you can omit this field in the components under the `components` folder

BJordi replied:
Great!! Thank you very much for the explanation! :smiley::+1:

mcallegari10 commented on src/views/NavBar.vue:
```diff
@@ -0,0 +1,49 @@
+</script>
+
+<style lang="scss" scoped >
+@import "../scss/commons/colors.scss";
```
No need for the extension

Commit:
MINOR FIXES ASKED FOR CR

- Moved NavBar to the components folder.
- Remove unnecessary empty line.
- Remove unnecessary component's name in NavBar.

Commit:
FIXES IN IMPORTS

- Remove aliases.
- Remove extensions of scss files.
- Changed height in main container

mcallegari10 commented on src/components/NavBar.vue:
```diff
@@ -0,0 +1,48 @@
+</script>
+
+<style lang="scss" scoped >
+@import "../scss/commons/colors.scss";
```
No need for the extension
